lib/sorcery/engine.rb

```diff
@@ -9,18 +9,12 @@ class Engine < Rails::Engine
 
     # TODO: Should this include a modified version of the helper methods?
     initializer 'extend Controller with sorcery' do
-      # FIXME: on_load is needed to fix Rails 6 deprecations, but it breaks
-      #        applications due to undefined method errors.
-      # ActiveSupport.on_load(:action_controller_api) do
-      if defined?(ActionController::API)
-        ActionController::API.send(:include, Sorcery::Controller)
+      ActiveSupport.on_load(:action_controller_api) do
+        ActionController::API.include(Sorcery::Controller)
       end
 
-      # FIXME: on_load is needed to fix Rails 6 deprecations, but it breaks
-      #        applications due to undefined method errors.
-      # ActiveSupport.on_load(:action_controller_base) do
-      if defined?(ActionController::Base)
-        ActionController::Base.send(:include, Sorcery::Controller)
+      ActiveSupport.on_load(:action_controller_base) do
+        ActionController::Base.include(Sorcery::Controller)
```

Comment:
This solution is confirmed as breaking for existing applications, although an alternative that does work is on the v1 rework.

Reply:
Do you mean #227 there? Or does this change cause other issues?

Reply:
Per FIXME, changing to on_load causes existing applications to generate undefined method errors. This has been pretty consistent every time we've tried switching.

Reply:
Are there any other [Sorcery/sorcery issues](https://github.com/Sorcery/sorcery/issues) related to switching to the `on_load` syntax other than #227? If yes, could you please point me to them?

Reply:
@athix [you say](https://github.com/Sorcery/sorcery/pull/255#issuecomment-671629664) that the deprecation warning from autoloading in Rails 6 (#209) has been fixed in Sorcery/sorcery-rework. When it is not going to be fixed in the main gem, then the question above is irrelevant and you can ignore it

Reply:
#132 is related

Reply:
> When it is not going to be fixed in the main gem, then the question above is irrelevant and you can ignore it

The rework (`sorcery-core`) will become the main gem once v1 ships, so in that sense it should be resolved in the main gem. As for back-porting it to the current `0.15.X` code-base, my free time is already completely consumed by the v1 rework so it's unlikely to happen until after `1.0.0` is live regardless.

Honestly, it would probably make sense for you to create a temporary fork with your changes here and use that until v1 is released. `0.15` shouldn't be receiving any updates in the meantime, so you won't have to worry about pulling down upstream changes until v1 goes live and you can cut over to that version.

Reply:
Ok, got it. Thank you for the advice @athix